lib/resource-timing.js

```diff
@@ -0,0 +1,228 @@
+/*!
+ * Copyright 2015 Comcast Cable Communications Management, LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+ (function(root, factory) {
+  if(typeof define === 'function' && define.amd) {
+    // For Require.js
+    define('surfnperf/resource-timing', ['surfnperf'], factory);
+  } else if(typeof exports === 'object') {
+    // For Browserify
+    module.exports = factory(require('surfnperf'));
+  } else {
+    // Browser global if not using Require.js or Browserify
+    root.surfnperfRT = factory(root.surfnperf);
+  }
+}(this, function(surfnperf) {
+  /**
```

Comment:
you can remove the Date.now() shim, as these Resource Timing functions don't need it